test/GenomeSet_basic_test.py

```diff
@@ -1,200 +1,217 @@
 
-    assert res["set_info"][1] == set_name
-    assert "KBaseSearch.GenomeSet" in res["set_info"][2]
+    return {
+        "obj": result,
+        "set_name": set_name,
+        "set_type": API_CLASS.set_type(),
+        # the class fills in the missing description field
+        "set_description": "",
+        "n_items": 0,
+    }
 
-    # test get of that object
-    d1 = set_api_client.get_genome_set_v1(context, {"ref": res["set_ref"]})[0]
-    assert "data" in d1
-    assert "info" in d1
-    assert len(d1["info"]) == INFO_LENGTH
-    assert "KBaseSearch.GenomeSet" in res["set_info"][2]
 
-    assert d1["data"]["description"] == "my kbasesearch genome set"
-    assert len(d1["data"]["elements"]) == n_items
+def test_save_genome_set(genome_set) -> None:
+    check_save_set_output(**genome_set)
 
-    elements = d1["data"]["elements"]
-    assert genome_refs[0] in elements
-    assert genome_refs[1] in elements
 
-    genome_2 = elements.get(genome_refs[1])
-    assert "ref" in genome_2
-    assert genome_2.get("ref") == genome_refs[1]
-    assert genome_2["metadata"]["test_metadata"] == "metadata"
+def test_save_empty_genome_set(empty_genome_set) -> None:
+    check_save_set_output(**empty_genome_set)
 
 
-def test_save_and_get_of_empty_set(
-    set_api_client: SetAPI, context: dict[str, str | list], ws_id: int
+def test_save_kbasesearch_genome(
+    kbase_search_genome_set: dict[str, Any],
+    genome_refs: list[str],
+    set_api_client: SetAPI,
+    context: dict[str, Any],
 ) -> None:
-    set_name = "nada_set"
-    set_description = "nothing to see here"
-
-    # create the set object
-    set_data = {"description": set_description, "items": []}
-    # test a save
-    res = set_api_client.save_genome_set_v1(
-        context,
-        {
-            "data": set_data,
-            "output_object_name": set_name,
-            "workspace_id": ws_id,
-        },
-    )[0]
-    assert "set_ref" in res
-    assert "set_info" in res
-    assert len(res["set_info"]) == INFO_LENGTH
-
-    assert res["set_info"][1] == set_name
-    assert "item_count" in res["set_info"][10]
-    assert res["set_info"][10]["item_count"] == "0"
+    res = kbase_search_genome_set["obj"]
+    check_save_set_output(
+        kbase_search_genome_set["obj"],
+        set_name=kbase_search_genome_set["set_name"],
+        set_type=kbase_search_genome_set["set_type"],
+    )
 
     # test get of that object
     d1 = set_api_client.get_genome_set_v1(context, {"ref": res["set_ref"]})[0]
     assert "data" in d1
     assert "info" in d1
-    assert len(d1["info"]) == INFO_LENGTH
-    assert "item_count" in d1["info"][10]
-    assert d1["info"][10]["item_count"] == "0"
-
-    assert d1["data"]["description"] == set_description
-    assert len(d1["data"]["items"]) == 0
-
-    d2 = set_api_client.get_genome_set_v1(
-        context, {"ref": res["set_ref"], INC_ITEM_INFO: 1}
-    )[0]
+    is_info_object(
```

Comment:
more annoying bespoke tests because the KBaseSearch.GenomeSet object is structured differently to the KBaseSet version.